src/Drivers/Jibit/JibitClient.php

```diff
@@ -0,0 +1,283 @@
+     * @var string
+     */
+    public $baseUrl;
+
+
+    public function __construct($apiKey, $secretKey, $baseUrl, $cachePath)
+    {
+        $this->baseUrl = $baseUrl;
+        $this->apiKey = $apiKey;
+        $this->secretKey = $secretKey;
+        $this->cache = new JibitFileCache($cachePath, 'jibit', '.cache');
+    }
+
+    /**
+     * Request payment
+     *
+     * @param int $amount
+     * @param string $referenceNumber
+     * @param string $userIdentifier
+     * @param string $callbackUrl
+     * @param string $currency
+     * @param null $description
+     * @param $additionalData
+     * @return bool|mixed|string
+     * @throws \Shetabit\Multipay\Exceptions\PurchaseFailedException
+     */
+    public function paymentRequest($amount, $referenceNumber, $userIdentifier, $callbackUrl, $currency = 'IRR', $description = null, $additionalData = null)
+    {
+        $this->generateToken();
+
+        $data = [
+            'additionalData' => $additionalData,
+            'amount' => $amount,
+            'callbackUrl' => $callbackUrl,
+            'clientReferenceNumber' => $referenceNumber,
+            'currency' => $currency,
+            'userIdentifier' => $userIdentifier,
+            'description' => $description,
+        ];
+
+        return $this->callCurl('/purchases', $data, true);
+    }
+
+    /**
+     *
+     * Get order by ID
+     * @param $id
+     * @return bool|mixed|string
+     * @throws \Shetabit\Multipay\Exceptions\PurchaseFailedException
+     */
+    public function getOrderById($id)
+    {
+        return  $this->callCurl('/purchases?purchaseId=' . $id, [], true, 0, 'GET');
+    }
+
+    /**
+     * Generate token
+     *
+     * @param bool $isForce
+     * @return string
+     * @throws \Shetabit\Multipay\Exceptions\PurchaseFailedException
+     */
+    private function generateToken($isForce = false)
+    {
+        $this->cache->eraseExpired();
+
+        if ($isForce === false && $this->cache->isCached('accessToken')) {
+            return $this->setAccessToken($this->cache->retrieve('accessToken'));
+        } elseif ($this->cache->isCached('refreshToken')) {
+            $refreshToken = $this->refreshTokens();
+
+            if ($refreshToken !== 'ok') {
+                return $this->generateNewToken();
+            }
+        } else {
+            return $this->generateNewToken();
+        }
+
+        throw new \Shetabit\Multipay\Exceptions\PurchaseFailedException('Token generation encountered an error.');
+    }
+
+    private function refreshTokens()
+    {
+        $data = [
+            'accessToken' => str_replace('Bearer ', '', $this->cache->retrieve('accessToken')),
+            'refreshToken' => $this->cache->retrieve('refreshToken'),
+        ];
+
+        $result = $this->callCurl('/tokens/refresh', $data, false);
+
+        if (empty($result['accessToken'])) {
+            throw new \Shetabit\Multipay\Exceptions\PurchaseFailedException('Refresh token encountered an error.');
```

Comment:
please import the exception and then use it. don't need to use fully qualified namespace+exception name each time.

Reply:
it's addressed in the next commit.